packages/admin-ui/src/checkbox/checkbox-input.tsx

```diff
@@ -1,37 +1,39 @@
-import { useEffect, useRef } from 'react'
-import type { CheckboxOptions as AriakitCheckboxOptions } from 'ariakit/checkbox'
+import type { Ref } from 'react'
+import React, { forwardRef, useEffect, useRef } from 'react'
 import { Checkbox as AriakitCheckbox } from 'ariakit/checkbox'
-import { createComponent, useElement } from '@vtex/admin-ui-react'
 import { useForkRef } from '@vtex/admin-ui-hooks'
 
-import * as style from './checkbox.style'
+import { cx } from '@vtex/admin-ui-core'
+import { checkboxTheme } from './checkbox.css'
 
-export const CheckboxInput = createComponent<
-  typeof AriakitCheckbox,
-  CheckboxInputOptions
->((props) => {
-  const { error = false, ref: htmlRef, state, ...htmlProps } = props
+export const CheckboxInput = forwardRef(function CheckboxInput(
+  props: CheckboxInputProps,
+  ref: Ref<HTMLInputElement>
+) {
+  const { error = false, state, className = '', ...htmlProps } = props
 
-  const ref = useRef<HTMLInputElement>(null)
+  const forkedRef = useRef<HTMLInputElement>(null)
```

Comment:
Out of curiosity, what's the concept behind forkedRef?

Reply:
It basically merges two refs. Here is used because we need it for the indeterminate state work